agent/api/task/task.go

```diff
@@ -3687,3 +3687,33 @@ func (task *Task) HasActiveContainers() bool {
 	}
 	return false
 }
+
+// IsManagedDaemonTask will check if a task is a non-stopped managed daemon task
+// TODO: We'll probably want to also clean up all of the STOPPED managed daemon task from
+func (task *Task) IsManagedDaemonTask() (string, bool) {
+	task.lock.RLock()
+	defer task.lock.RUnlock()
+
+	// We'll want to obtain the last known non-stopped managed daemon task to be saved into our task engine.
+	// There can be an edge case where the task hasn't been progressed to RUNNING yet.
+	taskStatus := task.KnownStatusUnsafe
+	if !task.IsInternal || taskStatus >= apitaskstatus.TaskStopped {
+		return "", false
+	}
+
+	for _, c := range task.Containers {
+		if c.IsManagedDaemonContainer() {
+			imageName := c.GetImageName()
+			return imageName, true
```

Comment:
if a managed daemon in future has more than a single container we'll need to figure out which container is THE container.  This might be an essential container, or else we'll need to enforce a one-daemon-container per daemon task rule.  We might want to add a TODO to track this as a task-level field in future.  
https://github.com/aws/amazon-ecs-agent/blob/master/agent/engine/daemonmanager/daemon_manager_linux.go#L131C3-L131C103
The daemons are currently very specifically focused on a single image as base.

Reply:
added